packages/berry-pnpify/sources/NodeModulesFS.ts

```diff
@@ -0,0 +1,266 @@
+import { CreateReadStreamOptions, CreateWriteStreamOptions } from '@berry/fslib';
+import { NodeFS, PosixFS, FakeFS, WriteFileOptions }         from '@berry/fslib';
+
+import fs                                                    from 'fs';
+
+import { NodePathResolver }                                  from './NodePathResolver';
+import { PnPApiLoader }                                      from './PnPApiLoader';
+import { PnPApiLocator }                                     from './PnPApiLocator';
+
+export type NodeModulesFSOptions = {
+  baseFs?: FakeFS
+};
+
+export class NodeModulesFS extends FakeFS {
+  private readonly baseFs: FakeFS;
+  private readonly pathResolver: NodePathResolver;
+
+  constructor({baseFs = new PosixFS(new NodeFS())}: NodeModulesFSOptions = {}) {
+    super();
+
+    this.baseFs = baseFs;
+    this.pathResolver = new NodePathResolver({
+      apiLoader: new PnPApiLoader({ watch: (fs as any).watch.bind(fs) }),
+      apiLocator: new PnPApiLocator({ existsSync: baseFs.existsSync.bind(baseFs) })
+    });
+  }
+
+  getRealPath() {
+    return '/';
+  }
+
+  getBaseFs() {
+    return this.baseFs;
+  }
+
+  private resolveFilePath(p: string): string {
+    const pnpPath = this.pathResolver.resolvePath(p);
+    if (!pnpPath.resolvedPath) {
+      throw new Error(`ENOENT: no such file or directory, stat '${p}'`)
```

Comment:
You need to use `Object.assign` to set the `code` member (maybe this should be put in an helper in the `fslib` package):

https://github.com/yarnpkg/berry/blob/master/packages/berry-fslib/sources/ZipFS.ts#L268